src/Microsoft.DotNet.Build.CloudTestTasks/DownloadFromAzure.cs

```diff
@@ -191,5 +207,14 @@ public override bool Execute()
                 clientThrottle.Release();
             }
         }
+
+        private void AddPathTooLongEntry(string blob, string filename)
+        {
+            var pathTooLongItem = new Microsoft.Build.Utilities.TaskItem(blob);
+            pathTooLongItem.SetMetadata("DestinationPath", filename);
+            pathTooLongItem.SetMetadata("DestinationPathLength", filename.Length.ToString());
+            pathTooLongEntries.Add(pathTooLongItem);
+            Log.LogWarning($"Unable to download blob '{blob}' as its final path exceeded the allowed length. This may cause problems if it was needed.");
```

Comment:
Will this cause the task to succeed if some files were unable to download? I think this task should still fail when this happens, just not on the first time it encounters this problem so it has the chance to log every path that fails.

Is there a scenario you have in mind where the caller would want to continue building but alter its behavior based on the items it was unable to download?

Reply:
The task already succeeds if there are blobs that have a directory-like name. Which implied having some blobs that fail to download is acceptable behavior. I can cause it to error out instead  (That's what happens currently, an exception is logged and the task fails).

If there are no cases where it makes sense to continue the process if a blob is not downloaded, maybe it makes sense to change the behavior of the directory-like blobs as well?

Reply:
Ah, now I see what you were saying in the description. I'm not sure why that was implemented as a warning in the first place--@MattGal may know more.

Reply:
@dagood we can make it an error but whoever was passing junk to this method will break; There's no case that I know of where there was actually a blob like this, it was intended, rather the Items being fed to it were malformed and one ended in a /.  

I'm fine making that an error instead of a warning;  I believe all known offenders of this have fixed their code anyways.

Reply:
In order to avoid breaking existing misbehaving consumers, I can maintain the current behavior for the ones that log a warning, and make the new PathTooLongException blobs return error.